api/board/README.md

```diff
@@ -55,10 +55,18 @@ The `address` property is the operating-system-specific representation for that
 
 <a name="pinmode">
 The `mode` property can take the following values:
-- `"input"` for digital input (GPIO). The pin value can be 0 or 1.
-- `"output"` for digital output (GPIO). The pin value can be 0 or 1.
-- `"analog"` for analog input (AIO) that is converted to digital value.
+- `"digital-input"` for digital input (GPIO). The pin value can be 0 or 1.
+- `"digital-output"` for digital output (GPIO). The pin value can be 0 or 1.
+- `"analog-input"` for analog input (AIO) that is converted to digital value.
+- `"analog-output"` for analog output (AIO) that is converted from digital value.
 - `"pwm"` for PWM analog output.
+- `"uart-rx"` for serial receive pin.
+- `"uart-tx"` for serial transmit pin.
+- `"i2c-scl"` for I2C clock.
+- `"i2c-scl"` for I2C data.
+- `"spi-sclk"` for SPI clock.
+- `"spi-mosi"` for SPI Master Out Slave In.
+- `"spi-miso"` for SPI Master In Slave Out.
```

Comment:
There are 2 possibilities we expose SPI.
1) Specify and SPI `bus` number/index (SCLK+MISO+MOSI), and then a `device` number/index to address the slave (SS) during a `transceive()` operation. The bus number is board specific, and implementation maps it to the physical pin configuration and the MOSI, MISO, SCLK pins are exposed (we could do without the latter).
2) Specify an array of 3+x `Pin` objects, where x >= 1 and denote the SS pins. This is also board specific, but it's not hidden behind bus number and device number.